services/core/processing/ingress.go

```diff
@@ -0,0 +1,43 @@
+package processing
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+
+	"sensorbucket.nl/sensorbucket/pkg/pipeline"
+)
+
+type IngressDTO struct {
+	TracingID  uuid.UUID `json:"tracing_id"`
+	PipelineID uuid.UUID `json:"pipeline_id"`
+	OwnerID    int64     `json:"owner_id"`
+	Payload    []byte    `json:"payload"`
+	CreatedAt  time.Time `json:"created_at"`
+}
+
+func CreateIngressDTO(pipeline uuid.UUID, ownerID int64, payload []byte) IngressDTO {
+	return IngressDTO{
+		TracingID:  uuid.New(),
+		CreatedAt:  time.Now(),
+		PipelineID: pipeline,
+		OwnerID:    ownerID,
+		Payload:    payload,
+	}
+}
+
+func TransformIngressDTOToPipelineMessage(dto IngressDTO, pl *Pipeline) (*pipeline.Message, error) {
+	pipelineMessage := pipeline.Message{
+		ID:            dto.TracingID.String(),
```

Comment:
is TracingID validated somewhere before String is called?

Reply:
The JSON Unmarshal in Transport should throw an error if an invalid UUID is set in the IngressDTO. All we can do here is validate that it is not a default 00000-000-000-0000 UUID